ftest/src/test/java/org/arquillian/reporter/GreeterReportVerifier.java

```diff
@@ -0,0 +1,229 @@
+            .isEqualTo(ReporterCoreKey.GENERAL_TEST_CLASS_CONFIGURATION_REPORT);
+
+        assertThat(testMethodReports).size().isEqualTo(2);
+    }
+
+    @Test
+    public void verify_test_class_configuration_report_exists_with_contents() {
+        List<TestClassReport> testClassReports = executionReport.getTestSuiteReports().get(0).getTestClassReports();
+        TestClassReport testClassReport = testClassReports.get(0);
+        ConfigurationReport configurationReport = testClassReport.getConfiguration();
+        Report deploymentReport = configurationReport.getSubReports().get(0);
+
+        assertThatReport(configurationReport)
+            .hasName(ReporterCoreKey.GENERAL_TEST_CLASS_CONFIGURATION_REPORT)
+            .hasNumberOfEntries(0)
+            .hasNumberOfSubReports(1);
+
+        verify_deployment_configuration_sub_report(deploymentReport);
+    }
+
+    private void verify_deployment_configuration_sub_report(Report deploymentReport) {
+        Report nestedDeploymentReport = (Report) deploymentReport.getSubReports().get(0);
+
+        assertThat(deploymentReport.getName())
+            .isEqualTo(new UnknownStringKey("deployments"));
+        assertThat(deploymentReport.getEntries()).size().isEqualTo(0);
+        assertThat(deploymentReport.getSubReports()).size().isEqualTo(1);
+
+        assertThatReport(nestedDeploymentReport)
+            .hasName(ArquillianCoreKey.DEPLOYMENT_IN_TEST_CLASS_REPORT)
+            .hasNumberOfEntries(4)
+            .hasEntryContainingKeys(ArquillianCoreKey.DEPLOYMENT_IN_TEST_CLASS_NAME,
+                ArquillianCoreKey.ARCHIVE_NAME_OF_DEPLOYMENT,
+                ArquillianCoreKey.ORDER_OF_DEPLOYMENT,
+                ArquillianCoreKey.PROTOCOL_USED_FOR_DEPLOYMENT)
+            .hasNumberOfSubReports(0);
+    }
+
+    @Test
+    public void verify_test_method_reports_exists_with_contents() {
+        List<TestClassReport> testClassReports = executionReport.getTestSuiteReports().get(0).getTestClassReports();
+        TestClassReport testClassReport = testClassReports.get(0);
+        List<TestMethodReport> testMethodReports = testClassReport.getTestMethodReports();
+
+        assertThat(testMethodReports).size().isEqualTo(2);
+
+        List<StringKey> testMethodReportNameList = Arrays.asList(new UnknownStringKey("run_client_test"),
+            new UnknownStringKey("should_create_greeting"));
+
+        for (TestMethodReport testMethodReport : testMethodReports) {
+            assertThat(testMethodReport.getName()).isIn(testMethodReportNameList);
+            assertThat(testMethodReport.getStatus()).isEqualTo(TestResult.Status.PASSED);
+            assertThatTestMethodReport(testMethodReport)
+                .hasNumberOfEntries(3)
+                .hasEntryContainingKeys(ArquillianCoreKey.TEST_METHOD_OPERATES_ON_DEPLOYMENT,
+                    ArquillianCoreKey.TEST_METHOD_RUNS_AS_CLIENT,
+                    ArquillianCoreKey.TEST_METHOD_REPORT_MESSAGE)
```

Comment:
These values as well are implemented in `ArquillianCoreReporterLifecycleManager` and are based on event triggers.